src/relax/transform/simplify_norm_inference.cc

```diff
@@ -0,0 +1,182 @@
+  return _sinfo.value();
+}
+
+Expr ExpandToMatchInput(Expr data, int ndim, Array<Integer> axes) {
+  axes = GetOrderedPositiveAxes(axes, ndim);
+  Array<Integer> expand_axes;
+  for (int i = 0, j = 0; i < ndim; ++i) {
+    if (j < static_cast<int>(axes.size()) && i == axes[j]->value) {
+      ++j;
+    } else {
+      expand_axes.push_back(i);
+    }
+  }
+  return expand_dims(data, expand_axes);
+}
+
+/*!
+ * \brief The common part for unpacking normalization
+ * \param dev The Expr of data - mean (may avoid duplicate computation)
+ * \param var The Expr of variance
+ * \param gamma The Expr of gamma, skip scaling if not defined.
+ * \param beta The Expr of beta, skip shifting if not defined.
+ * \param axes The axes to reduce.
+ * \param eps The epsilon value.
+ * \param ndim The ndim of the input data.
+ * \param dtype The dtype of the input data.
+ * \return The Expr of the unpacked normalization.
+ */
+Expr SimplifyCommonNorm(Expr dev, Expr var, Optional<Expr> gamma, Optional<Expr> beta,
+                        Array<Integer> axes, float eps, int ndim, DataType dtype) {
+  // output = (x - mean) / sqrt(var + epsilon) * gamma + beta
+  Expr epsilon = MakeConstantScalar(eps, dtype);
+  Expr sqrt_var = sqrt(add(var, epsilon));
+  Expr out = divide(dev, sqrt_var);
+
+  if (gamma.defined()) {
+    out = multiply(out, ExpandToMatchInput(gamma.value(), ndim, axes));
+  }
+  if (beta.defined()) {
+    out = add(out, ExpandToMatchInput(beta.value(), ndim, axes));
+  }
+  return out;
+}
+
+std::pair<Expr, Expr> ComputeDevVar(Expr data, Array<Integer> axes) {
+  Expr mean_expr = mean(data, axes, /*keepdims=*/true);
+  Expr dev = subtract(data, mean_expr);
+  Expr var = mean(multiply(dev, dev), axes, /*keepdims=*/true);
+  return std::make_pair(dev, var);
+}
+
+Expr SimplifyBatchNorm(const CallNode* call) {
+  auto attrs = call->attrs.as<BatchNormAttrs>();
+  ICHECK_NOTNULL(attrs);
+  Optional<Expr> gamma = attrs->scale ? call->args[1] : Optional<Expr>(NullOpt);
+  Optional<Expr> beta = attrs->center ? call->args[2] : Optional<Expr>(NullOpt);
+  TensorStructInfo sinfo = MatchTensorStructInfo(call->args[0]);
+  Expr moving_mean = ExpandToMatchInput(call->args[3], sinfo->ndim, {attrs->axis});
+  Expr moving_var = ExpandToMatchInput(call->args[4], sinfo->ndim, {attrs->axis});
+  Expr dev = subtract(call->args[0], moving_mean);
+  return SimplifyCommonNorm(dev, moving_var, gamma, beta, {attrs->axis}, attrs->epsilon,
+                            sinfo->ndim, sinfo->dtype);
+}
+
+Expr SimplifyLayerNorm(const CallNode* call) {
+  auto attrs = call->attrs.as<LayerNormAttrs>();
+  ICHECK_NOTNULL(attrs);
+  Optional<Expr> gamma = attrs->scale ? call->args[1] : Optional<Expr>(NullOpt);
+  Optional<Expr> beta = attrs->center ? call->args[2] : Optional<Expr>(NullOpt);
+  auto res = ComputeDevVar(call->args[0], attrs->axes);
+  const Expr& dev = res.first;
+  const Expr& var = res.second;
```

Comment:
Not sure if you would like to try the following if we go with layernorm simplification.
```suggestion
  auto [dev, var] = ComputeDevVar(call->args[0], attrs->axes);
```